### ISSUE TITLE
Reader.deviceSoftwareVersion is sometimes returning its version as "fw-config-ksid"

### ISSUE THREAD

**Issue by dpoineau** [stripe/stripe-terminal-ios#199]
## Summary
We've noticed intermittently that the card readers we use (M2, Chipper2x, WisePad3) will return "fw-config-ksid" when calling reader.deviceSoftwareVersion

## Code to reproduce
I'm not sure which case is happening when the firmware version is returned as "fw-config-ksid", but we call reader.deviceSoftwareVersion from a few different places:
- reader:didReportBatteryLevel:status:isCharging
- Terminal.shared.connectBluetoothReader completion handler
- Terminal.shared.connectedReader (when checking if a reader is already connected before starting a payment)

## iOS version
Various versions

## Installation method
Cocoapods

## SDK version
2.13.0

## Other information
I understand from the documentation that `reader.deviceSoftwareVersion` can return `nil` in some cases, but is it intentional that it sometimes also returns this `fw-config-ksid` string instead of either `nil` or an actual version?

**Comment by dpoineau** [comment]
@bric-stripe So I was able to find a good example, and in this case I'm pretty sure it's coming from the `reader:didReportBatteryLevel:status:isCharging` call specifically (because this card reader actually has low battery ~11%, so I can see we're getting the `didReportBatteryLevel` call every couple of minutes - does it make this callback more frequently when low battery?)

`Serial #: WPC323104000151`

The most recent time (as of this writing) that it got that version string was today, around `19:53:18 UTC`. Some more recent times (in case it's helpful) were:
```
19:52:00
19:49:39
19:47:33
```

**Comment by bric-stripe** [comment]
thanks for that extra info 👍 

> does it make this callback more frequently when low battery

No, I wouldn't expect that. The SDK will refresh the level/charging state on the host app becoming active again. I'll take a look to see if anything else odd is going on there though.

**Comment by bric-stripe** [comment]
Thanks again for the serial and timing. With that I was able to review some logging on our side and am pretty sure I found the issue. From what I can tell one of the battery polling requests takes longer than expected to get a response from the reader and times out. This shouldn't show up as a problem anywhere except now this `deviceSoftwareVersion` will be reset back to the not-terribly-helpful `fw-config-ksid` you're seeing. We should be able to get this fixed in our end-of-January release.
